src/controllers/discord/oauth.ts

```diff
@@ -0,0 +1,107 @@
+import type { Request, Response } from 'express';
+import axios from 'axios';
+import qs from 'qs';
+import env from '../../utils/env';
+import database from '../../services/database';
+import { fetchUser } from '../../operations/user';
+import { ActionFeedback, DiscordFeedbackCode } from '../../types';
+import { decrypt } from '../../utils/helpers';
+import { isLoginAllowed } from '../../middlewares/settings';
+import type { DiscordAuthorization, DiscordToken, DiscordTokenRequest, DiscordAuthorizationData } from './discordApi';
+import logger from '../../utils/logger';
+
+const redirect = (response: Response, statusCode: DiscordFeedbackCode) => {
+  response
+    .status(302)
+    .setHeader(
+      'Location',
+      `${env.front.website}/dashboard/account?action=${ActionFeedback.DISCORD_OAUTH}&state=${statusCode}`,
+    );
+  response.end();
+};
+
+export default [
+  isLoginAllowed,
+  async (request: Request<{}, {}, {}, DiscordAuthorization>, response: Response) => {
+    try {
+      const { error, code, state } = request.query;
+
+      // At that point, an error is most likely because the user denied the permission access
+      if (typeof error !== 'undefined') return redirect(response, DiscordFeedbackCode.ERR_OAUTH_DENIED);
+
+      // If there is no code with no error, it is probably a bad request !
+      if (typeof code !== 'string' || typeof state !== 'string')
+        return redirect(response, DiscordFeedbackCode.ERR_BAD_REQUEST);
+
+      // Retrieve user from `state`
+      const executorId = decrypt(Buffer.from(state, 'base64'));
+      const executorPromise = fetchUser(executorId);
+
+      // We use the [authorization_code to fetch the access token]
+      // (https://discord.com/developers/docs/topics/oauth2#authorization-code-grant)
+      const tokenRequest = axios.post<DiscordToken>(
+        'https://discord.com/api/v9/oauth2/token',
+        qs.stringify(<DiscordTokenRequest>{
+          client_id: env.discord.client,
+          client_secret: env.discord.token,
+          code,
+          grant_type: 'authorization_code',
+          redirect_uri: env.discord.oautback,
+        }),
+        {
+          headers: {
+            'Content-Type': 'application/x-www-form-urlencoded',
+          },
+          timeout: 1000,
+        },
+      );
+      const [{ data: token }, executor] = await Promise.all([tokenRequest, executorPromise]);
```

Comment:
```suggestion
      const [{ data: tokenResponse }, executor] = await Promise.all([tokenRequest, executorPromise]);
```
Dans un soucis de lisibilité